connection.js

```diff
@@ -352,28 +346,83 @@ class Connection {
             return { reject: true, msg: "This bot accepts Unranked games only. " };
         }
 
-        // for square board sizes only
-        if (notification.width != notification.height && !config.allow_all_sizes && !config.allow_custom_sizes) {
-            conn_log("board was not square, rejecting challenge");
-            return { reject: true, msg: "In your selected board size " + notification.width + "x" + notification.height + " (width x height), Board was not square, please choose a square board size (same width and height, for example 9x9 or 19x19). " };
+        // for all the allowed_family options below (timecontrols, speeds, komi, boardsizes) 
+        // we need to add a "family guard" 
+        // && config.familyranked for ranked games 
+        // && config.familyunranked for unranked games
+        // else the allowed_ is always false and always rejects
+
+        /******** begining of BOARDSIZES *********/
+        // for square board sizes only //
+        /* if not square*/
+        if (notification.width != notification.height && !config.allow_all_sizes && !config.allow_custom_sizes && !config.boardsizeranked && !config.boardsizeunranked) {
+            conn_log("board was not square, not allowed");
+            return { reject: true, msg: "Your selected board size " + notification.width + "x" + notification.height + " (width x height), is not square, not allowed, please choose a square board size (same width and height, for example 9x9 or 19x19). " };
         }
 
-        if (!config.allowed_sizes[notification.width] && !config.allow_all_sizes && !config.allow_custom_sizes) {
-            conn_log("square board size " + notification.width + "x" + notification.height + " is not an allowed size, rejecting challenge");
-            return { reject: true, msg: "Board size " + notification.width + "x" + notification.height + " is not allowed, please choose one of the allowed square board sizes (same width and height, for example if allowed boardsizes are 9,13,19, it means you can play only 9x9 , 13x13, and 19x19), these are the allowed square board sizes : " + config.boardsize };
+        if (notification.width != notification.height && !config.allow_all_sizes_ranked && !config.allow_custom_sizes_ranked && notification.ranked) {
```

Comment:
Similar as `===`, use `!==` instead of `!=`

Reply:
similar answer as my previous comment, i totally agree too :+1: